### PR TITLE
add wrapper for finding monthly localized covariance

### PR DESCRIPTION
**Pull request by veni-vidi-vici-dormivi**
Add a wrapper around `find_localized_empirical_covariance` to produce a covariance matrix for every month to use for MESMER_M

 - [x] Tests added
 - [x] Fully documented, including `CHANGELOG.rst`

### CODE REVIEW

File: mesmer/stats/_localized_covariance.py [hunk 1]
```diff
@@ -84,7 +84,7 @@ def find_localized_empirical_covariance(
     Parameters
     ----------
     data : xr.DataArray
-        2D DataArray with with to calculate the covariance for.
+        2D DataArray with data to calculate the covariance for.
```
Comment (mathause) [inline mesmer/stats/_localized_covariance.py:87]: Nice catch!

File: mesmer/stats/_localized_covariance.py [hunk 2]
```diff
@@ -144,6 +144,65 @@ def find_localized_empirical_covariance(
...
+):
+    """determine localized empirical covariance by cross validation for each month
+
+    Parameters
+    ----------
+    data : xr.DataArray
+        2D DataArray with monthly data to calculate the covariance for.
+    weights : xr.DataArray
+        Weights for the individual samples.
+    localizer : dict of DataArray```
```
Comment (mathause) [inline mesmer/stats/_localized_covariance.py:158]:
```suggestion
    localizer : dict of DataArray
```

File: mesmer/stats/_localized_covariance.py [hunk 3]
```diff
@@ -144,6 +144,65 @@ def find_localized_empirical_covariance(
...
+    Notes
+    -----
+    Runs a k-fold cross validation if ``k_folds`` is smaller than the number of samples
+    and a leave-one-out cross validation otherwise.
+    """
+    localized_ecov = []
+
+    for mon in range(1, 13):
+        data_mon = data.groupby(f"{dim}.month")[mon]
+        weights_mon = weights.groupby(f"{dim}.month")[mon]
```
Comment (mathause) [inline mesmer/stats/_localized_covariance.py:190]:
```suggestion
    data_grouped = data.groupby(f"{dim}.month")
    weights_grouped = weights.groupby(f"{dim}.month")

    for mon in range(1, 13):
        data_mon = data_grouped[mon]
        weights_mon = weights_grouped[mon]
```

**Review by mathause — COMMENTED** [review]
LGTM

File: mesmer/stats/_localized_covariance.py [hunk 4]
```diff
@@ -144,6 +144,65 @@ def find_localized_empirical_covariance(
...
+
+        localized_ecov.append(
+            find_localized_empirical_covariance(
+                data_mon,
+                weights_mon,
```
Comment (mathause) [inline mesmer/stats/_localized_covariance.py:195]:
```suggestion
                data_grouped[mon],
                weights_grouped[mon],
```

File: mesmer/stats/_localized_covariance.py (hunk 3 above)
Comment (veni-vidi-vici-dormivi) [reply mesmer/stats/_localized_covariance.py]: Good point, and prettier anyways

File: tests/unit/test_localized_covariance.py [hunk 5]
```diff
@@ -98,6 +99,63 @@ def test_find_localized_empirical_covariance():
...
 
+@pytest.mark.filterwarnings("ignore:First element is local minimum.")
+def test_find_localized_empirical_covariance_monthly():
+
+    n_samples = 20 * 12
+    n_gridpoints = 60
+    time = pd.date_range("2000-01-01", periods=n_samples, freq="MS")
+
+    data = get_random_data(n_samples, n_gridpoints)
+    data["samples"] = time
```
Comment (mathause) [inline tests/unit/test_localized_covariance.py:110]:
```suggestion
```

File: tests/unit/test_localized_covariance.py [hunk 6]
```diff
@@ -98,6 +99,63 @@ def test_find_localized_empirical_covariance():
...
+    data.assign_coords({"samples": time})
+
+    localizer = get_localizer_dict(n_gridpoints, as_dataarray=True)
+
+    weights = get_weights(n_samples)
+    weights["samples"] = time
```
Comment (mathause) [inline tests/unit/test_localized_covariance.py:116]:
```suggestion
```

File: tests/unit/test_localized_covariance.py [hunk 7]
```diff
@@ -98,6 +99,63 @@ def test_find_localized_empirical_covariance():
...
+        "shape": (12, n_gridpoints, n_gridpoints),
+    }
+
+    result = mesmer.stats.find_localized_empirical_covariance_monthly(
+        data, weights, localizer, dim="samples", k_folds=2
+    )
+
+    np.testing.assert_equal(result.localization_radius.values, np.zeros(12))
+    _check_dataarray_form(result.covariance, "covariance", **required_form)
+    _check_dataarray_form(result.covariance, "localized_covariance", **required_form)
```
Comment (mathause) [inline tests/unit/test_localized_covariance.py:131]:
```suggestion
    _check_dataarray_form(result.localized_covariance, "localized_covariance", **required_form)
```

File: tests/unit/test_localized_covariance.py [hunk 8]
```diff
@@ -98,6 +99,63 @@ def test_find_localized_empirical_covariance():
...
+
+    # ensure it works if data is transposed
+    result = mesmer.stats.find_localized_empirical_covariance_monthly(
+        data.T, weights, localizer, dim="samples", k_folds=3
+    )
+
+    np.testing.assert_equal(result.localization_radius.values, np.zeros(12))
+    _check_dataarray_form(result.covariance, "covariance", **required_form)
+    _check_dataarray_form(result.covariance, "localized_covariance", **required_form)
```
Comment (mathause) [inline tests/unit/test_localized_covariance.py:140]: dito

File: tests/unit/test_localized_covariance.py [hunk 9]
```diff
@@ -98,6 +99,63 @@ def test_find_localized_empirical_covariance():
...
+        dim="samples",
+        k_folds=3,
+        equal_dim_suffixes=(":j", ":i"),
+    )
+
+    required_form["required_dims"] = ("cell:j", "cell:i")
+
+    np.testing.assert_equal(result.localization_radius.values, np.zeros(12))
+    _check_dataarray_form(result.covariance, "covariance", **required_form)
+    _check_dataarray_form(result.covariance, "localized_covariance", **required_form)
```
Comment (mathause) [inline tests/unit/test_localized_covariance.py:156]: ...

File: mesmer/stats/_localized_covariance.py [hunk 10]
```diff
@@ -144,6 +144,65 @@ def find_localized_empirical_covariance(
...
+        localized_ecov.append(
+            find_localized_empirical_covariance(
+                data_grouped[mon],
+                weights_grouped[mon],
+                localizer,
+                dim=dim,
+                k_folds=k_folds,
+                equal_dim_suffixes=equal_dim_suffixes,
+            )
+        )
```
Comment (mathause) [inline mesmer/stats/_localized_covariance.py:200]:
```suggestion
        res = find_localized_empirical_covariance(
            data_grouped[mon],
            weights_grouped[mon],
            localizer,
            dim=dim,
            k_folds=k_folds,
            equal_dim_suffixes=equal_dim_suffixes,
        )
        localized_ecov.append(res)
```

nit but I find that a bit clearer

**Review by mathause — APPROVED** [review]

**Pull request merged**